code/game/gamemodes/gameticker.dm

```diff
@@ -649,18 +651,24 @@ var/datum/controller/gameticker/ticker
 			roles += player.mind.assigned_role
 	return roles
 
-/datum/controller/gameticker/proc/handle_lights()
-	var/list/discrete_areas = areas.Copy()
-	//Get department areas where there is a crewmember. This is used to turn on lights in occupied departments
+/datum/controller/gameticker/proc/handle_lights() //This is used to turn on lights in occupied departments
+	var/list/lit_up_areas = list()
+	for(var/mob/living/player in player_list)
+		lit_up_areas += get_department_areas(player)
```

Comment:
Unless my understanding of byond code is wrong, doesn't this add the area many times? Say there's 5 medbay crewmates, this list will have medbay's areas 5 times. This makes lit_up_areas overly large, and since you iterate over lit_up_area every single time a light switch is meant to be turned off, that's no bueno. 
This would be better made as an unordered_set (in c++ anyways), but I'm guessing byond doesn't have that. It would be better then to create a "filtered_lit_up_areas" between 657 and 658 which is lit_up_areas without duplicates and then use that in the following for loop.

Reply:
thanks for checking. I think i've resolved the issue